impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtBuilder.java

```diff
@@ -595,14 +595,8 @@ private String sign(final Payload payload, final Key key, final Provider provide
 
             // Next, b64 extension requires the raw (non-encoded) payload to be included directly in the signing input,
             // so we ensure we have an input stream for that:
-            if (payload.isClaims() || payload.isCompressed()) {
-                ByteArrayOutputStream claimsOut = new ByteArrayOutputStream(8192);
-                writeAndClose("JWS Unencoded Payload", payload, claimsOut);
-                payloadStream = Streams.of(claimsOut.toByteArray());
-            } else {
-                // No claims and not compressed, so just get the direct InputStream:
-                payloadStream = Assert.stateNotNull(payload.toInputStream(), "Payload InputStream cannot be null.");
-            }
+            payloadStream = toInputStream("JWS Unencoded Payload", payload);
```

Comment:
I'm sorry I didn't make this clear: the term `unencoded payload` only applies when the `b64` extension is in use, and that's only a JWS option.

So in this particular call (line 598), `JWS Unencoded Payload` is correct because it's only being used within the `} else { // b64` block starting on line 591.

Reply:
Looks good! Thank you! 🚀